### PR TITLE
#347 web toggle category

### LINKED ISSUE #347: Update web toggle to filter by category not type
Filtering entities via the web UI is done via the `type` field.

## Expected Behaviour

Entities should be filtered by category by matching `description` on category entities.

## Current Behaviour

See above.

## Possible Solution

Update web UI query and add `category` enum types.

### PR DESCRIPTION
**Pull request by wlthomson**
Fixes #347.

## Description

Untangles a bit of the category vs. type cludges. A bit clearer, but could still use a bit of clarification re. type mappings. Backend resolvers can only be fixed up so much, as need to retain support of existing mSupply query (until we can migrate everyone off it). 

### Documentation:

Tick below if one of the following applies:

1. Documentation has been added to cover changes introduced by this PR.
2. This PR requires no changes to documentation.

- [x] Documentation OK.

### Tests:

Tick below if one of the following applies:

1. Tests have been added to cover changes introduced by this PR.
2. This PR requires no new tests.

- [x] Tests OK.

### CODE REVIEW

**Review by katherine-sussol — APPROVED** [review]
Nice! Like the separation of categories from types 👍

**Comment by katherine-sussol** [comment]
Also just noticed the entity description prefixes have disappeared 🙀  was this intended (I can't see what change did it without digging deeper)? I think it's okay because you can kinda read it going down the levels on the page anyway but... bit less user friendly? 🤷‍♀️ . Maybe better for a separate issue if it bothers others.

Before (master):
image

After:
image

**Comment by wlthomson** [comment]
@katherine-sussol 

Hmm, I could be wrong but this might be being caused by something else. No changes here directly touch the `entity` query; and it's a bit weird that it's working identically with the exception of the description. For the earlier mSupply fix the descriptions at the same level as shown had to be recursively constructed due to the updated data format. Could this be the same issue? Only way I can think of this being the case is if you possibly didn't update the data until you changed branches?

File: unified-codes/apps/data-service/src/app/v1/types/DgraphDataSource.ts [hunk 1]
```diff
@@ -29,34 +30,32 @@ export class DgraphDataSource extends RESTDataSource {
...
+  }
+
+  private static getEntityCategoryString(categories: string[]) {
+    return JSON.stringify(categories.map(category => {
+      switch(category) {
+        case EEntityCategory.DRUG:
           return 'Drug';
-        case EEntityType.MEDICINAL_PRODUCT:
+        case EEntityCategory.MEDICINAL_PRODUCT:
+        case EEntityCategory.OTHER:
```
[the remark this answers is not in the document]
Comment (wlthomson) [reply unified-codes/apps/data-service/src/app/v1/types/DgraphDataSource.ts]: Good idea! 90fb13f46.

**Comment by katherine-sussol** [comment]
> @katherine-sussol
> 
> Hmm, I could be wrong but this might be being caused by something else. No changes here directly touch the `entity` query; and it's a bit weird that it's working identically with the exception of the description. For the earlier mSupply fix the descriptions at the same level as shown had to be recursively constructed due to the updated data format. Could this be the same issue? Only way I can think of this being the case is if you possibly didn't update the data until you changed branches?

Yeah, it didn't make any sense to me either, but just switching branches from master to the PR without reloading data or anything seems to show the different behaviour? 🤷  Tried switching back and forth a few times. Probably why it's better investigated in a separate issue though because I couldn't see any changes that would cause it (I looked around a fair bit too 🤔 )

**Comment by katherine-sussol** [comment]
It might be related to this haha:
```
        case 'unit_of_use':
          fullDescription =  `${entityDescription} ${description}`;
          break;
```

**Pull request merged**